### PR TITLE
Implement detail page of a flight

### PR DESCRIPTION
**Pull request by anhmht**
#68 
 - When click on a flight in the flight list -> show a modal contain the detail of a flight (FlightDetailPage.jsx) 
 - Create a list of tracking for each flight
 - Show button 'Kiểm tra' if status of the flight is ON_BOARD at the bottom of this modal
screencapture-localhost-3000-flights-2019-03-22-08_08_09

### CODE REVIEW

**Comment by hoadinh2010** [comment]
image
Move the button to the right side.

File: imports/ui/modules/flights/components/FlightDetailPage.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+class FlightDetailPage extends React.PureComponent {
+  static propTypes = {
+    flight: shape().isRequired,
+    isLoading: bool.isRequired,
+    trackings: arrayOf(shape()).isRequired,
+  }
+
+  state = {}
+
+  displayFlightStatus = (status) => {
```
Comment (hoadinh2010) [inline imports/ui/modules/flights/components/FlightDetailPage.jsx]: Should move the color of each  status to the status enum like: 
`export const FLIGHT_STATUSES_COLOR = {
  [FLIGHT_STATUSES.PREPARE]: 'red',
  [FLIGHT_STATUSES.ON_BOARD]: 'green',
  [FLIGHT_STATUSES.ARRIVED]: 'blue'....,
};`


ANd using that:

`<Tag color={FLIGHT_STATUSES_COLOR[FLIGHT_STATUSES.ON_BOARD}>{FLIGHT_STATUSES_TEXT[FLIGHT_STATUSES.ON_BOARD]}</Tag>;`

File: imports/ui/modules/flights/components/FlightDetailPage.jsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    if (status === FLIGHT_STATUSES.CHECKING) {
+      return <Tag color="cyan">{FLIGHT_STATUSES_TEXT[FLIGHT_STATUSES.CHECKING]}</Tag>;
+    }
+    if (status === FLIGHT_STATUSES.DONE) {
+      return <Tag color="green">{FLIGHT_STATUSES_TEXT[FLIGHT_STATUSES.DONE]}</Tag>;
+    }
+    return undefined;
+  }
+
+  displayTrackingStatus = (status) => {
```
Comment (hoadinh2010) [inline imports/ui/modules/flights/components/FlightDetailPage.jsx]: The same above problem

File: imports/ui/modules/flights/components/FlightDetailPage.jsx (hunk 1 above)
Comment (anhmht) [reply imports/ui/modules/flights/components/FlightDetailPage.jsx]:
```
export const TRACKING_STATUSES_COLOR = {
  [TRACKING_STATUSES.IN_TRANSIT]: 'gold',
  [TRACKING_STATUSES.US_DELIVERED]: 'geekblue',
  [TRACKING_STATUSES.ON_HOLD]: 'orange',
  [TRACKING_STATUSES.PACKED]: 'cyan',
  [TRACKING_STATUSES.ON_BOARD]: 'magenta',
  [TRACKING_STATUSES.VN_DELIVERED]: 'volcano',
  [TRACKING_STATUSES.DOOR_DELIVERED]: 'green',
};
```

```
export const FLIGHT_STATUSES_COLOR = {
  [FLIGHT_STATUSES.PREPARE]: 'gold',
  [FLIGHT_STATUSES.ON_BOARD]: 'geekblue',
  [FLIGHT_STATUSES.ARRIVED]: 'orange',
  [FLIGHT_STATUSES.CHECKING]: 'cyan',
  [FLIGHT_STATUSES.DONE]: 'green',
};
```
```
<Tag color={FLIGHT_STATUSES_COLOR[flight.status]}>{FLIGHT_STATUSES_TEXT[flight.status]}</Tag>
```

```
<Tag color={TRACKING_STATUSES_COLOR[status]}>{TRACKING_STATUSES_TEXT[status]}</Tag>
```

File: imports/ui/modules/flights/components/FlightDetailPage.jsx (hunk 2 above)
Comment (anhmht) [reply imports/ui/modules/flights/components/FlightDetailPage.jsx]: fix as u said

**Comment by anhmht** [comment]
screencapture-localhost-3000-flights-2019-03-22-18_50_41

**Pull request merged**